InterviewService.swift

```diff
@@ -0,0 +1,71 @@
+
+import Foundation
+
+enum Mode {
+    case preprod
+    case prod
+}
+
+enum TypeService {
+    case Network
+    case appSetting
+    case mock
+    case calculate
+}
+
+enum Flow {
+    case firstFlow, secondFlow, thirdFlow
+}
+
+protocol NewServiceInput {
+    func getType() -> Any?
+    func calculateValue()
+}
+
+final class NewService: NSObject {
+    var typeService: TypeService
+    var mode: Mode
+    
+    //Get data from database
+    let model: [Model] = [Model(id: 1, numberCard: 123, company: "", state: ""),
```

Comment:
[Model], no need to mark the type, yet could be beneficial for the build time performance optimasation in some cases. Not here, I suppose.

Better inject from the outside.

Move injection into initialiser.